src/scripts/components/App/App.js

```diff
@@ -0,0 +1,431 @@
+        food: getRandomCoordinates(),
+        soundAction: 'playEat',
+      })
+      this.enlargeSnake();
+      this.increaseSpeed();
+    }
+  }
+
+  enlargeSnake() {
+    let newSnake = [...this.state.snakeDots];
+    newSnake.unshift([]);
+    this.setState({
+      snakeDots: newSnake
+    })
+  }
+
+  increaseSpeed() {
+    if (this.state.speed > 10) {
+      this.setState({
+        speed: this.state.speed - 10
+      })
+    }
+  }
+
+  onGameOver() {
+    clearInterval(this.state.intervalId);
+    this.setState({
+      isGameStart: false,
+      isScoreHidden: false,
+      soundAction: 'stop'
+    });
+  }
+
+  toggleShowSettings = () => {
+    this.setState(({ isSettingsShow }) => {
+      return {
+        isSettingsShow: !isSettingsShow
+      };
+    });
+  }
+
+  togglePlayMusic = () => {
+    this.setState(({ isPlayMusic }) => {
+      return {
+        isPlayMusic: !isPlayMusic
+      };
+    });
+  }
+
+  togglePlaySoundEffect = () => {
+    this.setState(({ isPlaySoundEffect }) => {
+      return {
+        isPlaySoundEffect: !isPlaySoundEffect
+      };
+    });
+  }
+
+  toggleSpeed = () => {
+    this.setState( () => {
+      let newSpeedRate = document.querySelector("#speed").value;
+      let newSpeed = 100 * newSpeedRate - (this.state.snakeDots.length * 10 - 20);
+      return {
+        startSpeedRate: newSpeedRate,
+        startSpeed: 100 * newSpeedRate,
+        speedRate: newSpeedRate,
+        speed: newSpeed
+      };
+    });
+  }
+
+  setMusicVolume = () => {
+    this.setState( () => {
+      return {
+        musicVolume: document.querySelector("#musicVolume").value
+      };
+    });
+  }
+
+  setSoundEffectVolume = () => {
+    this.setState( () => {
+      return {
+        soundEffectVolume: document.querySelector("#soundEffectVolume").value
+      };
+    });
+  }
+
+  toggleShowHotkeys = () => {
+    this.setState(({ isHotkeysShow }) => {
+      return {
+        isHotkeysShow: !isHotkeysShow
+      };
+    });
+    console.log(this.state.isHotkeysShow);
+  }
+
+  render() {
+    const btnMusicText = this.state.isPlayMusic ? 'On' : 'Off';
+    const btnSoundEffectText = this.state.isPlaySoundEffect ? 'On' : 'Off';
+
+    return (
+        <div className="app-wrapper">
+      <div className="main-block">
+
+        <Sound action={this.state.soundAction}
+          isPlayMusic={this.state.isPlayMusic}
+          isPlaySoundEffect={this.state.isPlaySoundEffect}
+          musicVolume={this.state.musicVolume}
+          soundEffectVolume={this.state.soundEffectVolume} />
+
+        <div className="buttons-area">
+
+          <div className="buttonsMain-area">
+            <ButtonPauseGame
+              text={this.state.isGamePause ? " Resume game" : " Pause game"}
+              isHidden={!this.state.isGameStart || this.state.isSettingsShow}
+              gameAction={this.state.isGamePause ? this.resumeGame : this.pauseGame}/>
+            <ButtonTryAgain
+              text=' Try again'
```

Comment:
Please move all your textes to separate constant file
Resume game
Pause game
Try again